src/OpenTelemetry.Instrumentation.AspNet/Implementation/HttpInListener.cs

```diff
@@ -76,23 +67,47 @@ private void OnStartActivity(Activity activity, HttpContext context)
             var request = context.Request;
             var requestValues = request.Unvalidated;
 
-            // see the spec https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md
+            // see the spec https://github.com/open-telemetry/semantic-conventions/blob/v1.24.0/docs/http/http-spans.md
             var path = requestValues.Path;
             activity.DisplayName = path;
 
-            if (request.Url.Port == 80 || request.Url.Port == 443)
+            var url = request.Url;
+            activity.SetTag(SemanticConventions.AttributeServerAddress, url.Host);
+            activity.SetTag(SemanticConventions.AttributeServerPort, url.Port);
+            activity.SetTag(SemanticConventions.AttributeUrlScheme, url.Scheme);
+
+            var originalHttpMethod = request.HttpMethod;
+            var normalizedHttpMethod = this.requestDataHelper.GetNormalizedHttpMethod(originalHttpMethod);
+            activity.SetTag(SemanticConventions.AttributeHttpRequestMethod, normalizedHttpMethod);
+
+            if (originalHttpMethod != normalizedHttpMethod)
```

Comment:
nit: Move this to a separate function similar to https://github.com/open-telemetry/opentelemetry-dotnet/blob/eb2dc443f4d92d91177c762e00738cfc76a1c4ca/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs#L200

Reply:
3a108ff6bd34455099d320015c8b16bd9d062227